### DISCUSSION TITLE
Different CustomVector6 Viewing Options

### DISCUSSION THREAD

**Discussion by Willy791** [ScanMountGoat/ssbh_editor#166] [Ideas]
I've noticed that while working with CustomVector6 in SSBH, the parameters are treated as an XYZW vector where X and Y are the horizontal and vertical scale of the UVs, and Z and W correspond to the translation. This is extremely helpful, although I've noticed the game treats it a little differently.

From my experience, when working with CV6, I've noticed that Z and W act moreso as end points, where over the course of 60 frames, the UVs linearly translate by however much the Z and W values are (where Z is X and W is Y). Essentially, instead of being strict translations, they're instructions for a consistent, perpetual translation. It's extremely noticeable in things such as Pirate Ship's sea, which uses a relatively simple shader with a custom CV6 value to allow for the illusion of movement.

My question is this: do you think it might at all be possible for implementation of a feature where CV6's Z and W values approximate for speed instead of translation? It'd be a massive help for mods with animated UV scroll aspects, as it'd allow for a much easier time estimating UV scroll speeds. Thanks in advance!

-Willy

**Comment by ScanMountGoat** [comment]
It's probably multiplying the UV translation by the elapsed time. This would explain the "speed" you observed. I don't have plans for when the next release will be, but I'll try and look into this.
